tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -664,35 +574,18 @@ async def federated_reduce(self, arg):
           'Expected 3 elements in the `federated_reduce()` argument tuple, '
           'found {}.'.format(len(arg.internal_representation)))
 
-    # Encrypt client tensors
-    pk_a, sk_a = await self._trusted_aggregator_generate_keys()
-    enc_clients_arg = await self._encrypt_client_tensors(arg, pk_a)
-    val_type = enc_clients_arg.type_signature
-    val = enc_clients_arg.internal_representation
-    # Store original client dtype before encryption for
-    # future decryption
-    orig_clients_dtype = arg.type_signature[0].member.dtype
+    # NOTE it will be supplied to the intrinsic strategy
+    agg_cls_channel = Channel(
+        sender_placement=placement_literals.CLIENTS,
+        receiver_placement=placement_literals.AGGREGATORS,
+        is_encrypted=True)
 
-    py_typecheck.check_type(val_type, computation_types.FederatedType)
-    item_type = val_type.member
     zero_type = arg.type_signature[1]
     op_type = arg.type_signature[2]
 
-    py_typecheck.check_type(val, list)
     aggr = self._get_child_executors(placement_literals.AGGREGATORS, index=0)
 
-    aggregands = await asyncio.gather(
-        *[self._move(v, item_type, aggr) for v in val])
-
-    # Decrypt tensors moved to the server before applying reduce
-    # In the future should be decrypted by the Trusted aggregator
-    aggregands_decrypted = []
-    for item in aggregands:
-      item_key_zipped = await self._zip_val_key(item, sk_a,
-                                                placement_literals.AGGREGATORS)
-      decrypted_tensor = await self._decrypt_tensors_on_aggregator(
-          item_key_zipped, orig_clients_dtype)
-      aggregands_decrypted.append(decrypted_tensor.internal_representation[0])
+    aggregands = await self._move(arg, aggr, self, agg_cls_channel)
```

Comment:
I think this is a bug -- the `self` you're passing as an arg should probably be something like `source = self._get_child_executors(placement_literals.CLIENTS)`

one nice thing about handling Channels in the `IntrinsicStrategy.__init__` is that you'd no longer need to pass the channel explicitly here, the channel can be inferred a combination of `self`, `aggr`, and `source`